src/main/java/gg/projecteden/nexus/features/commands/staff/admin/JobsCommand.java

```diff
@@ -0,0 +1,121 @@
+package gg.projecteden.nexus.features.commands.staff.admin;
+
+import gg.projecteden.nexus.Nexus;
+import gg.projecteden.nexus.framework.commands.models.CustomCommand;
+import gg.projecteden.nexus.framework.commands.models.annotations.Aliases;
+import gg.projecteden.nexus.framework.commands.models.annotations.Path;
+import gg.projecteden.nexus.framework.commands.models.annotations.Permission;
+import gg.projecteden.nexus.framework.commands.models.events.CommandEvent;
+import gg.projecteden.nexus.models.scheduledjobs.ScheduledJobs;
+import gg.projecteden.nexus.models.scheduledjobs.ScheduledJobsService;
+import gg.projecteden.nexus.models.scheduledjobs.common.AbstractJob;
+import gg.projecteden.nexus.models.scheduledjobs.common.AbstractJob.JobStatus;
+import gg.projecteden.nexus.models.scheduledjobs.common.RetryIfInterrupted;
+import gg.projecteden.nexus.models.scheduledjobs.jobs.Test2Job;
+import gg.projecteden.nexus.models.scheduledjobs.jobs.TestJob;
+import gg.projecteden.nexus.utils.Tasks;
+import gg.projecteden.utils.TimeUtils.Time;
+import lombok.NonNull;
+import org.reflections.Reflections;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static gg.projecteden.nexus.models.scheduledjobs.common.AbstractJob.getNextExecutionTime;
+
+@Aliases("job")
+@Permission("group.admin")
+public class JobsCommand extends CustomCommand {
+	private static final ScheduledJobsService service = new ScheduledJobsService();
+	private static final ScheduledJobs jobs = service.get0();
+
+	public JobsCommand(@NonNull CommandEvent event) {
+		super(event);
+	}
+
+	static {
+		checkInterrupted();
+		processor();
+		rescheduler();
+	}
+
+	private static void checkInterrupted() {
+		try {
+			final List<AbstractJob> interrupted = new ArrayList<>(jobs.get(JobStatus.RUNNING));
+			if (!interrupted.isEmpty()) {
+				for (AbstractJob job : interrupted) {
+					if (job.getClass().getAnnotation(RetryIfInterrupted.class) != null) {
+						job.setStatus(JobStatus.PENDING);
+						Nexus.warn("[Jobs] Found interrupted job, retrying: " + job);
+					} else {
+						job.setStatus(JobStatus.INTERRUPTED);
+						Nexus.severe("[Jobs] Found interrupted job: " + job);
+					}
+				}
+
+				service.save(jobs);
+			}
+		} catch (Exception ex) {
+			ex.printStackTrace();
+		}
+	}
+
+	private static void processor() {
+		Tasks.repeat(0, Time.SECOND, () -> {
```

Comment:
seems like previous implementations of tasks were async, but this is sync, is that intentional?

Reply:
I didn't feel the need to make any of the current tasks asynchronous, they are small enough, or in one case spawn their own async task. Will keep an eye on it tho